[UpstreamMergeAutomation] Added details to PR description + clean up

Shreejit-03 opened the pull request:
# Implementation

- Switched from rt_target_IP to ssh_connection for direct RT target access.
- Updated automation_conf.json and JsonConfig to use ssh_connection.
- Refactored build_and_test to pass ssh_connection into install_and_test_image.
- Normalized merge report key to "Push and PR".
- Added ni/rtos team tag in PR descriptions for quicker reviewer visibility.
- Adjusted clean script path to clean.core-feeds_and_core-images.sh (skip_merge scenario compatible).
- Updated README to reflect SSH changes.

# Justification
[AB#3174387](https://dev.azure.com/ni/DevCentral/_workitems/edit/3174387)

# Testing
[Azure DevOps pipeline](https://dev.azure.com/ni/DevCentral/_build/results?buildId=16328631&view=logs&j=ad00c9d0-876c-57ea-0fe7-ad9dc076256b) run completed successfully.

PRs were auto-created:
- https://github.com/ni/meta-selinux/pull/17
- https://github.com/ni/meta-openembedded/pull/100
- https://github.com/ni/pyrex/pull/19
- https://github.com/ni/openembedded-core/pull/172

Mail report was generated and sent successfully.

@pratheekshasn 
@rajendra-desai-ni

rajendra-desai-ni commented on scripts/dev/upstream_merge/README.md:
```diff
@@ -97,8 +97,8 @@ The script is user configurable and uses `automation_conf.json` to define variou
-- **`rt_target_IP`**:
-  The IP address or hostname of the RT target where images will be installed and tested via SSH.
+- **`ssh_connection`**:
+  The SSH connection of the RT target where images will be installed.
```
You can give a usage example something like: [username]@[hostname or IP address]

Shreejit-03 replied:
I have made those changes.

pratheekshasn commented on scripts/dev/upstream_merge/upstream_merge_and_test.py:
```diff
@@ -158,6 +158,7 @@ def push_submodules_and_create_PRs(
+    current_dir = os.getcwd()
     push_and_pr_results = {}
     for git_obj, (status, message) in list(merge_report.items()):
         if status == 0 and message is not None:
```
Isn't `os.chdir(git_obj.local_repo)` meant to `cd` to the absolute path of the submodule? If that's the case, you won't need to keep switching back to the `current_dir` at the end of every iteration.

Shreejit-03 replied:
After the final PR is created, the working directory should return to the nilrt directory.

pratheekshasn commented:
You might want to change the PR title and the commit message. Do add some more detail in the commit message after the title.

Shreejit-03 committed:
upstream_merge: Improve merge automation scripts

Rename rt_target_IP to ssh_connection to accept a
full SSH connection string (user@host) instead of
just an IP address.

Improve merge report formatting:
- Use descriptive status labels instead of OK/ERRORS
- Uppercase layer names in reports
- Add ERROR DETAILS and SUMMARY section headers

Improve PR creation:
- Auto-generate PR titles from submodule names
- Add testing checklist and team mention to PR body
- Remove hardcoded pr_title parameter
- Restore working directory after PR creation

Fix build steps:
- Pass build args to build_desirable_packages
- Fix clean script name to match actual filename
- Add name_only option to git_diff

Clear default email addresses in sample config and
remove stale contact from README.

Signed-off-by: Shreejit-03 <shreejit.c@emerson.com>

chaitu236 reviewed (approved):
Took a quick look, seems ok.
